tests/Aspire.Confluent.Kafka.Tests/ConsumerConfigurationTests.cs

```diff
@@ -102,14 +102,111 @@ public void ConnectionNameWinsOverConfigSection(bool useKeyed)
+    {
+        var builder = Host.CreateEmptyApplicationBuilder(null);
+
+        var key = useKeyed ? "messaging" : null;
+        builder.Configuration.AddInMemoryCollection([
+            new KeyValuePair<string, string?>("ConnectionStrings:messaging", CommonHelpers.TestingEndpoint),
+            new KeyValuePair<string, string?>(ProducerConformanceTests.CreateConfigKey("Aspire:Confluent:Kafka:Consumer", key, "Config:GroupId"), "unused")
+        ]);
+
+        bool configureBuilderIsCalled = false, configureSettingsIsCalled = false;
+
+        Action act =
+            (useKeyed, useConfigureSettings, useConfigureBuilder, useConfigureBuilderWithServiceProvider) switch
+            {
+                (true, true, false, false) => () => builder.AddKeyedKafkaConsumer<string, string>("messaging",
+                    configureSettings: ConfigureSettings),
+                (true, true, true, false) => () => builder.AddKeyedKafkaConsumer<string, string>("messaging",
+                    configureSettings: ConfigureSettings, configureBuilder: ConfigureBuilder),
+                (true, true, false, true) => () => builder.AddKeyedKafkaConsumer<string, string>("messaging",
+                    configureSettings: ConfigureSettings, configureBuilder: ConfigureBuilderWithServiceProvider),
+                (true, false, false, false) => () =>
+                    builder.AddKeyedKafkaConsumer<string, string>("messaging"),
+                (true, false, true, false) => () =>
+                    builder.AddKeyedKafkaConsumer<string, string>("messaging", configureBuilder: ConfigureBuilder),
+                (true, false, false, true) => () =>
+                    builder.AddKeyedKafkaConsumer<string, string>("messaging",
+                        configureBuilder: ConfigureBuilderWithServiceProvider),
+                (false, true, false, false) => () => builder.AddKafkaConsumer<string, string>("messaging",
+                    configureSettings: ConfigureSettings),
+                (false, true, true, false) => () => builder.AddKafkaConsumer<string, string>("messaging",
+                    configureSettings: ConfigureSettings, configureBuilder: ConfigureBuilder),
+                (false, true, false, true) => () => builder.AddKafkaConsumer<string, string>("messaging",
+                    configureSettings: ConfigureSettings, configureBuilder: ConfigureBuilderWithServiceProvider),
+                (false, false, false, false) => () =>
+                    builder.AddKafkaConsumer<string, string>("messaging"),
+                (false, false, true, false) => () =>
+                    builder.AddKafkaConsumer<string, string>("messaging", configureBuilder: ConfigureBuilder),
+                (false, false, false, true) => () =>
+                    builder.AddKafkaConsumer<string, string>("messaging",
+                        configureBuilder: ConfigureBuilderWithServiceProvider),
```

Comment:
This is just a suggestion. What do you think of formatting it a little differently to help readability? like https://gist.github.com/radical/b168cca070fbe1fae392ae8d99480d34

This helps to see all the overloads being covered, and you can easily see that keyed/non-keyed both have been covered. And if one was missing it would be easy to spot.

Reply:
Looks nice. I formatted both producer and consumer parameterized tests that way.